src/modules/users/users.service.ts

```diff
@@ -54,16 +60,27 @@ export class UsersService {
     await this.isEmailExist(email);
 
     const user = await this.prisma.user.create({ data: { email, hashedPassword } });
+    await this.addRoleToUser(user.id, ROLES.USER);
     return user;
   }
 
-  public async getAll(): Promise<User[]> {
-    const users = await this.prisma.user.findMany({ include: { roles: { include: { role: true } } } });
+  public async getAll(params: QueryParamsDto): Promise<PaginatedResponse<UserWithRelationsWithoutPassword>> {
+    const result = await super.getPaginatedResult<UserWithRoles>({
+      model: this.prisma.user,
+      params,
+      include: { roles: { include: { role: true } }, profile: { include: { author: true } } },
+    });
 
-    return users.map((user) => ({
-      ...user,
-      roles: user.roles.map((userRole) => userRole.role.name),
-    }));
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
+    const safeUsers = result.items.map(({ hashedPassword, ...rest }) => rest);
```

Comment:
We actually have the pattern for this 🙃 
```suggestion
    const safeUsers = result.items.map(({ _hashedPassword, ...rest }) => rest);
```

Reply:
This isn't possible due to the destructuring